    -Fazer análise exploratória nos dados
        =Title - relevante para postagens
        =Score - relevante para postagens - com base no impacto, número de comentários
        =Id - id único para postagens / comentários
        =Url - relevante para postagens - url do tópico de postagem
        =Commns_num - relevante para a postagem - número de comentários para esta postagem
        =Created - data de criação
        =Body - relevante para postagens / comentários - texto da postagem ou comentário
        =Timestamp - timestamp

In [9]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np
import plotly.express as px
from plotly.subplots import make_subplots
rddVm = pd.read_csv("D:/Programação/Python/Arquivos Externos/reddit_vm.csv")

display(rddVm)

display(rddVm.info())


title  score       id  \
0     Health Canada approves AstraZeneca COVID-19 va...      7   lt74vw   
1     COVID-19 in Canada: 'Vaccination passports' a ...      2   lsh0ij   
2     Coronavirus variants could fuel Canada's third...      6   lohlle   
3     Canadian government to extend COVID-19 emergen...      1   lnptv8   
4     Canada: Pfizer is 'extremely committed' to mee...      6   lkslm6   
...                                                 ...    ...      ...   
1561                                            Comment      2  elqz2wa   
1562                                            Comment      0  elqybv1   
1563                                            Comment     -1  elqy6em   
1564                                            Comment      1  elqwqiv   
1565                                            Comment      2  elqnxtl   

                                                    url  comms_num  \
0     https://www.canadaforums.ca/2021/02/health-can...          0   
1     https://www.canadaforums.ca/2021/02/covid-19-i...          1   
2     https://www.canadaforums.ca/2021/02/coronaviru...          0   
3     https://www.canadaforums.ca/2021/02/canadian-g...          0   
4     https://www.canadaforums.ca/2021/02/canada-pfi...          0   
...                                                 ...        ...   
1561                                                NaN          0   
1562                                                NaN          0   
1563                                                NaN          0   
1564                                                NaN          0   
1565                                                NaN          0   

           created                                               body  \
0     1.614400e+09                                                NaN   
1     1.614316e+09                                                NaN   
2     1.613887e+09                                                NaN   
3     1.613796e+09                                                NaN   
4     1.613468e+09                                                NaN   
...            ...                                                ...   
1561  1.556203e+09  https://en.wikipedia.org/wiki/Thiomersal\n\nEt...   
1562  1.556203e+09                                    Excellent. Why?   
1563  1.556203e+09                                          M o n e y   
1564  1.556202e+09                        What is thimerosal made of?   
1565  1.556196e+09  Compounds aren't forms of elements. That's lit...   

                timestamp  
0     2021-02-27 06:33:45  
1     2021-02-26 07:11:07  
2     2021-02-21 07:50:08  
3     2021-02-20 06:35:13  
4     2021-02-16 11:36:28  
...                   ...  
1561  2019-04-25 17:41:04  
1562  2019-04-25 17:33:18  
1563  2019-04-25 17:31:45  
1564  2019-04-25 17:16:41  
1565  2019-04-25 15:34:45  

[1566 rows x 8 columns]

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1566 entries, 0 to 1565
Data columns (total 8 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   title      1566 non-null   object 
 1   score      1566 non-null   int64  
 2   id         1566 non-null   object 
 3   url        461 non-null    object 
 4   comms_num  1566 non-null   int64  
 5   created    1566 non-null   float64
 6   body       1196 non-null   object 
 7   timestamp  1566 non-null   object 
dtypes: float64(1), int64(2), object(5)
memory usage: 98.0+ KB


None

In [10]:
##Limpeza dos dados

rddVm = rddVm.drop(['url','id'], axis=1)
print(rddVm.shape)
display(rddVm.info())


(1566, 6)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1566 entries, 0 to 1565
Data columns (total 6 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   title      1566 non-null   object 
 1   score      1566 non-null   int64  
 2   comms_num  1566 non-null   int64  
 3   created    1566 non-null   float64
 4   body       1196 non-null   object 
 5   timestamp  1566 non-null   object 
dtypes: float64(1), int64(2), object(3)
memory usage: 73.5+ KB


None

In [11]:
titulo = rddVm['title'].unique()

for col in rddVm:
    display(f"Quantidade de {col}: {len(rddVm[col].unique())}")


'Quantidade de title: 462'

'Quantidade de score: 43'

'Quantidade de comms_num: 36'

'Quantidade de created: 1566'

'Quantidade de body: 1175'

'Quantidade de timestamp: 1566'

In [12]:
#separando os dados em duas partes, parte 1)posts 2)comentários
coments = rddVm[rddVm['title'] == 'Comment']
post = rddVm[rddVm['title'] != 'Comment']

display(f"{len(coments)}")
display(f"{len(post)}")

display(f"Valor maximo {coments['score'].max()}")
display(f"Valor minimo {coments['score'].min()}")

display(f"Valor maximo {post['score'].max()}")
display(f"Valor minimo {post['score'].min()}")



'1105'

'461'

'Valor maximo 12'

'Valor minimo -12'

'Valor maximo 1192'

'Valor minimo 0'

In [34]:
#analisando a parte das postagens
pst = pd.DataFrame()
l1 = []
l2 = []

maxPostScore = post['score'].max()
maxPostComen = post['comms_num'].max()

for i in range(0,maxPostScore):
    l1.append(f"Nº score igual {i}")
    l2.append(len(post[post['score'] == i]))
pst['Descrição'] = l1
pst['Qtd'] = l2

pst  = pst[pst['Qtd'] != 0]


gf1 = px.bar(post, x=post.index , y=post['score'],title="Postagem X Score")
gf2 = px.bar(post, x=post.index , y=post['comms_num'], title="Postagem X Número de comnentários")

gf1.show()
gf2.show()

display(pst)

gfPst = px.bar(pst, x=pst.index, y=pst['Qtd'], title="Qantidade de socre por indice")
gfPst.show()

com = pd.DataFrame()
lc1 = []
lc2 = []

for i in range(0,maxPostComen):
    lc1.append(f"Nº comentarios igual a {i}")
    lc2.append(len(post[post['comms_num'] == i]))
com['Descrição'] = lc1
com['Qtd'] = lc2

com  = com[com['Qtd'] != 0]

display(com)

gfCom = px.bar(com, x=com.index, y=com['Qtd'], title="Qantidade de comentarios por indice")
gfCom.show()


Descrição  Qtd
0    Nº score igual 0   48
1    Nº score igual 1   47
2    Nº score igual 2   49
3    Nº score igual 3   42
4    Nº score igual 4   40
5    Nº score igual 5   41
6    Nº score igual 6   24
7    Nº score igual 7   27
8    Nº score igual 8   21
9    Nº score igual 9   14
10  Nº score igual 10   20
11  Nº score igual 11   11
12  Nº score igual 12    7
13  Nº score igual 13    5
14  Nº score igual 14    7
15  Nº score igual 15   10
16  Nº score igual 16   12
17  Nº score igual 17    5
18  Nº score igual 18    4
19  Nº score igual 19    3
20  Nº score igual 20    2
21  Nº score igual 21    5
22  Nº score igual 22    3
24  Nº score igual 24    2
25  Nº score igual 25    3
26  Nº score igual 26    1
27  Nº score igual 27    1
29  Nº score igual 29    1
30  Nº score igual 30    1
31  Nº score igual 31    1
36  Nº score igual 36    1
40  Nº score igual 40    1
44  Nº score igual 44    1

Descrição  Qtd
0      Nº comentarios igual a 0   48
1      Nº comentarios igual a 1  143
2      Nº comentarios igual a 2   76
3      Nº comentarios igual a 3   38
4      Nº comentarios igual a 4   40
5      Nº comentarios igual a 5   21
6      Nº comentarios igual a 6   11
7      Nº comentarios igual a 7    8
8      Nº comentarios igual a 8    9
9      Nº comentarios igual a 9    7
10    Nº comentarios igual a 10    4
11    Nº comentarios igual a 11    7
12    Nº comentarios igual a 12    6
13    Nº comentarios igual a 13    3
15    Nº comentarios igual a 15    5
16    Nº comentarios igual a 16    3
17    Nº comentarios igual a 17    3
18    Nº comentarios igual a 18    2
19    Nº comentarios igual a 19    2
20    Nº comentarios igual a 20    3
21    Nº comentarios igual a 21    1
22    Nº comentarios igual a 22    3
24    Nº comentarios igual a 24    2
25    Nº comentarios igual a 25    1
27    Nº comentarios igual a 27    1
30    Nº comentarios igual a 30    1
31    Nº comentarios igual a 31    1
32    Nº comentarios igual a 32    3
34    Nº comentarios igual a 34    1
39    Nº comentarios igual a 39    1
46    Nº comentarios igual a 46    1
49    Nº comentarios igual a 49    2
55    Nº comentarios igual a 55    1
74    Nº comentarios igual a 74    1
166  Nº comentarios igual a 166    1

In [37]:
#analisando a parte de comentários
coments = coments[coments['score'] > 0]

gf3 = px.bar(coments, x=coments.index, y=coments['score'])

gf3.show()

maxScorCom = coments['score'].max()
scoCom = pd.DataFrame()
lsc1 = []
lsc2 = []

for i in range(0,maxScorCom):
    lsc1.append(f"Nº score igual {i}")
    lsc2.append(len(coments[coments['comms_num'] == i]))
scoCom['Descrição'] = lc1
scoCom['Qtd'] = lc2

scoCom = scoCom[scoCom['Qtd'] != 0]

display(scoCom)

gfScorCom = px.bar(scoCom, x=scoCom.index, y=scoCom['Qtd'], title="Qantidade de comentarios por indice")
gfScorCom.show()



Descrição  Qtd
0      Nº comentarios igual a 0   48
1      Nº comentarios igual a 1  143
2      Nº comentarios igual a 2   76
3      Nº comentarios igual a 3   38
4      Nº comentarios igual a 4   40
5      Nº comentarios igual a 5   21
6      Nº comentarios igual a 6   11
7      Nº comentarios igual a 7    8
8      Nº comentarios igual a 8    9
9      Nº comentarios igual a 9    7
10    Nº comentarios igual a 10    4
11    Nº comentarios igual a 11    7
12    Nº comentarios igual a 12    6
13    Nº comentarios igual a 13    3
15    Nº comentarios igual a 15    5
16    Nº comentarios igual a 16    3
17    Nº comentarios igual a 17    3
18    Nº comentarios igual a 18    2
19    Nº comentarios igual a 19    2
20    Nº comentarios igual a 20    3
21    Nº comentarios igual a 21    1
22    Nº comentarios igual a 22    3
24    Nº comentarios igual a 24    2
25    Nº comentarios igual a 25    1
27    Nº comentarios igual a 27    1
30    Nº comentarios igual a 30    1
31    Nº comentarios igual a 31    1
32    Nº comentarios igual a 32    3
34    Nº comentarios igual a 34    1
39    Nº comentarios igual a 39    1
46    Nº comentarios igual a 46    1
49    Nº comentarios igual a 49    2
55    Nº comentarios igual a 55    1
74    Nº comentarios igual a 74    1
166  Nº comentarios igual a 166    1

    - No código é possível perceber que há dois tipos de conteúdos que são os Posts e Comments
        ° Posts e Comments
            .Com isso foi separado o em duas partes a análise dos posts e dos commnts
            .Nos post percebe-se que há poucos posts que passam de 27 no score e de 39 comentários
            .Na pate de comentários nota-se que há muitos comnetários com score igual a 1 
        
        °° Outliers
            .Em ambas as partes há dados muito destoante onde no posts ha uma postagem que possui um score de 1192 e mesmo também possui 596 comentários 
            destoando assim dos outros posts.
            
            .Na parte dos comentários se encontra mais agrupado, com isso os comentário apresentam uma relevância semelhante, indicando que não houve 
            comentários que fossem exraodinário o sulficiente para destoar.